src/main/java/org/prebid/server/bidder/yieldlab/YieldlabBidder.java

```diff
@@ -231,6 +244,60 @@ private static String getConsentParameter(User user) {
         return ObjectUtils.defaultIfNull(consent, "");
     }
 
+    private static Map<String, String> extractDsaRequestParamsFromBidRequest(BidRequest request) {
+        return Optional.ofNullable(request.getRegs())
+            .map(Regs::getExt)
+            .map(ExtRegs::getDsa)
+            .map(YieldlabBidder::extractDsaRequestParamsFromDsaRegsExtension)
+            .orElseGet(Map::of);
+    }
+
+    private static Map<String, String> extractDsaRequestParamsFromDsaRegsExtension(final ExtRegsDsa dsa) {
+        final HashMap<String, String> dsaRequestParams = new HashMap<>();
+
+        if (dsa.getDsaRequired() != null) {
+            dsaRequestParams.put("dsarequired", dsa.getDsaRequired().toString());
+        }
+
+        if (dsa.getPubRender() != null) {
+            dsaRequestParams.put("dsapubrender", dsa.getPubRender().toString());
+        }
+
+        if (dsa.getDataToPub() != null) {
+            dsaRequestParams.put("dsadatatopub", dsa.getDataToPub().toString());
+        }
+
+        final List<ExtRegsDsaTransparency> dsaTransparency = dsa.getTransparency();
+        if (dsaTransparency != null && !dsaTransparency.isEmpty()) {
+            dsaRequestParams.put("dsatransparency", encodeTransparenciesAsString(dsaTransparency));
+        }
```

Comment:
`Stream.ofNullable(dsa.getTransparency()).flatMap(Collection::stream).map(this::encode(...)).ifPresent(transparency -> params.put(...));`

Reply:
I'm not sure I understand you here, but `Stream.ofNullable()` returns an empty stream for null objects, not an Optional.

Reply:
You're right, the simple `Optional.ofNullable(transparecy).filter(CollectionUtils::isNotEmpty).ifPresent(...);` will be enough

Reply:
I will leave this as it is too then since I would rather not use Optionals for null-checks.